cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/statements/expressions/DeclaredReferenceExpression.java

```diff
@@ -138,13 +139,18 @@ public void possibleSubTypesChanged(HasType src, List<HasType> root) {
     if (!TypeManager.isTypeSystemActive()) {
       return;
     }
+
+    // since we want to update the sub types, we need to exclude ourselves from the root, otherwise
+    // it won't work. What a weird and broken system!
+    root.remove(this);
```

Comment:
@KuechA 

Just FYI, this was one of those weird cases where I had to remove the node from `root`, but hopefully this will be obsolete in the future anyway.